src/transaction_builder.js

```diff
@@ -308,18 +307,18 @@ function buildFromInputData (input, scriptType, parentType, redeemScript, allowI
 
   switch (scriptType) {
     case 'pubkeyhash':
-      var pkhSignature = input.signatures[0].toScriptSignature(input.hashType)
+      var pkhSignature = bscript.signature.encode(input.signatures[0], input.hashType)
       scriptSig = bscript.pubKeyHashInput(pkhSignature, input.pubKeys[0])
       break
 
     case 'pubkey':
-      var pkSignature = input.signatures[0].toScriptSignature(input.hashType)
+      var pkSignature = bscript.signature.encode(input.signatures[0], input.hashType)
       scriptSig = bscript.pubKeyInput(pkSignature)
       break
 
     case 'multisig':
       var msSignatures = input.signatures.map(function (signature) {
-        return signature && signature.toScriptSignature(input.hashType)
+        return signature && bscript.signature.encode(signature, input.hashType)
       })
 
       // fill in blanks with OP_0
```

Comment:
This file outlines the real differences that users will feel.

IMHO it is an improvement,  but @jprichardson I'd be up to hear your opinion.
Less magic,  clear parameters and a functional style.

The only thing I don't like is that `.decode(signature).signature` is a thing.
Perhaps
``` js
{
 r: "...",
 s: "...",
 hashType: 0
}
```

Would be better